runbotics/runbotics-scheduler/src/queue/process-instance/process-instance.scheduler.service.ts

```diff
@@ -47,5 +52,21 @@ export class ProcessInstanceSchedulerService {
             BotWsMessage.TERMINATE,
             processInstanceId
         );
+
+        const activeProcessInstanceEvents = await this.processInstanceEventService.findAllActiveByProcessInstanceId(processInstanceId);
+
+        if (activeProcessInstanceEvents.length > 0) {
+            activeProcessInstanceEvents.forEach((event)=>{
+                const newProcessInstanceEvent = {
+                    ...event,
+                    status: ProcessInstanceEventStatus.TERMINATED,
+                };
+
+                this.processInstanceEventService.update(newProcessInstanceEvent)
+                    .catch((err) => {
+                        this.logger.error(`Process instance event update error: ${err}`);
+                    });
+            });
+        }
```

Comment:
calling this logic here won't work as we want it to. Move it to bot gateway method that receives process instance messages. If `terminate` status arrives then do smth more. Here we will make changes on DB before the process is terminated on bot

Reply:
additionally, make it guard if, reduce nesting if possible

Reply:
guard if has been added

Reply:
done